controls/radpersistenceframework/persistence-framework-allowed-types.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Allowed Types
+page_title: Allowed Types
+description: Register types that are allowed to be instantiated by the deserialization mechanism of RadPersistenceFramework.
+slug: persistence-framework-allowed-types
+tags: allowed types, types, persistence framework
+published: True
+position: 8
+---
+
+# Allowed Types
+
+When loading the persisted layout, the `RadPersistenceFramework` will instantiate the saved types only if they are allowed. This is controlled by the `PersistenceManager` class, which has predefined types that are allowed by default. You can also register other types by modifying its `AllowedTypes` collection or by using the `AllowTypes` extension method.
+
+#### __[C#] Allowing a type using the AllowedTypes collection of the PersistenceManager__
+{{region persistence-framework-allowed-types-0}}
+    PersistenceManager manager = new PersistenceManager();
+    manager.AllowedTypes.Add(typeof(SolidColorBrush));
+{{endregion}}
+
+#### __[VB.NET] Allowing a type using the AllowedTypes collection of the PersistenceManager__
+{{region persistence-framework-allowed-types-1}}
+    Dim manager As PersistenceManager = New PersistenceManager()
+    manager.AllowedTypes.Add(GetType(SolidColorBrush))
+{{endregion}}
+
+#### __[C#] Allowing a type using the AllowTypes extension method__
+{{region persistence-framework-allowed-types-2}}
+    PersistenceManager manager = new PersistenceManager().AllowTypes(typeof(SolidColorBrush));
+{{endregion}}
+
+#### __[VB.NET] Allowing a type using the AllowTypes extension method__
+{{region persistence-framework-allowed-types-3}}
+    Dim manager As PersistenceManager = New PersistenceManager().AllowTypes(GetType(SolidColorBrush))
+{{endregion}}
+
+In cases where internal WPF types need to be added to the AllowedTypes collection, you can obtain them at runtime and utilize the AllowTypes method. One such case is with the native WPF `Selector` class that assigns the `SelectedItems` property to a `SelectedItemCollection` type, which is internal. In this specific scenario, invoking the `AllowInputControls` method will resolve this, however, if you need to allow internal types, you can follow the approach from the following example:
```

Comment:
I think that this section with the internal types is less important than the information about the Telerik extension methods. I suggest moving the Telerik types section and making this a separate section like : ## Adding Internal Types